frontend/packages/dev-console/src/components/health-checks/AddHealthCheckPage.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { RouteComponentProps } from 'react-router-dom';
+import { PageHeading, FirehoseResource, Firehose } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { ServiceModel as KnativeServiceModel } from '@console/knative-plugin';
+import AddHealthCheckWrapper from './AddHealthCheckWrapper';
+
+type AddHealthCheckProps = RouteComponentProps<{
+  ns: string;
+  kind: string;
+  name: string;
+  containerName: string;
+}>;
+
+const AddHealthCheckPage: React.FC<AddHealthCheckProps> = ({ match }) => {
+  const { ns, kind, name, containerName } = match.params;
+  const resource: FirehoseResource[] = [
+    {
+      kind: kind === KnativeServiceModel.kind ? referenceForModel(KnativeServiceModel) : kind,
```

Comment:
You should be using a qualified reference for knative always so that you don't need to do these one off checks.